### PR TITLE
Coordinates

### PR DESCRIPTION
**Pull request by ParsonsRD**
This is a relatively major refactoring of the code, lots of variable names have been changed such that then are more understandable and the frames have been split into 2 file to differentiate between ground frames and angular systems as this two frames type never interact with one another.

Also a new coordinate representation have been added for systems that can be described as a 2d plane. Finally the Horizon system has been added, which is functionally equivalent to the AltAz system but allows us to convert to it directly from the camera frame.

### CODE REVIEW

**Comment by MaxNoe** [comment]
You got some minor markup errors in the docstrings. From travis:

```
/home/travis/build/cta-observatory/ctapipe/build/lib.linux-x86_64-3.5/ctapipe/coordinates/angular_frames.py:docstring of ctapipe.coordinates.HorizonFrame:14: WARNING: Bullet list ends without a blank line; unexpected unindent.

['TiltedGroundFrame(*args, **kwargs)', ':module: ctapipe.coordinates', '', 'Bases: :class:`astropy.coordinates.baseframe.BaseCoordinateFrame`', '', '', '', 'Tilted ground coordinate frame.  The tilted ground coordinate frame', 'is a cartesian system describing the 2 dimensional projected', 'positions of objects in a tilted plane described by', 'pointing_direction Typically this frame will be used for the', 'reconstruction of the shower core position', '', 'Frame attributes:', '* ``pointing_direction``', '  Alt,Az direction of the tilted reference plane', '', '.. rubric:: Attributes Summary', '', '.. autosummary::', '', '   ~TiltedGroundFrame.default_representation', '   ~TiltedGroundFrame.frame_attributes', '   ~TiltedGroundFrame.frame_specific_representation_info', '   ~TiltedGroundFrame.name', '   ~TiltedGroundFrame.pointing_direction', '', '', '', '', '', '', '', '', '', '', '', '.. rubric:: Attributes Documentation', '', '', '.. autoattribute:: default_representation', '.. autoattribute:: frame_attributes', '.. autoattribute:: frame_specific_representation_info', '.. autoattribute:: name', '.. autoattribute:: pointing_direction']:13: ERROR: Unexpected indentation.
```

File: ctapipe/coordinates/angular_frames.py [hunk 1]
```diff
@@ -85,26 +84,56 @@ class NominalFrame(BaseCoordinateFrame):
...
+    Horizon coordinate frame. Spherical system used to describe the direction
+    of a given position, in terms of the altitude and azimuth of the system. In
+    practice this is functionally identical as the astropy AltAz system, but this
+    implementation allows us to pass array pointing information, allowing us to directly
+    transform to the Horizon Frame from the Camera system.
+
+    The Following attributes are carried over from the telescope frame
+    to allow a direct transformation from the camera frame
+
+   * ``array_direction``
```
Comment (MaxNoe) [inline ctapipe/coordinates/angular_frames.py]: I think here a space is missing here before the `*`

Comment (ParsonsRD) [inline ctapipe/coordinates/angular_frames.py]: Ah great, I'm glad you found that, I was struggling to find the error :-).

**Comment by MaxNoe** [comment]
I think we use the astropy doc standards. They also provide a standardized way to document class attributes:

http://docs.astropy.org/en/stable/development/docrules.html#documenting-classes

File: ctapipe/coordinates/angular_frames.py [hunk 2]
```diff
@@ -85,26 +84,56 @@ class NominalFrame(BaseCoordinateFrame):
...
+    implementation allows us to pass array pointing information, allowing us to directly
+    transform to the Horizon Frame from the Camera system.
+
+    The Following attributes are carried over from the telescope frame
+    to allow a direct transformation from the camera frame
+
+    * ``array_direction``
+      Alt,Az direction of the array pointing
+    * ``pointing_direction``
+      Alt,Az direction of the telescope pointing
```
Comment (MaxNoe) [inline ctapipe/coordinates/angular_frames.py]: I think this should be

```
Attributes
-----------

array_direction: astropy.coordinates.FrameAttribute
    Alt, Az direction of the array pointing
...
```

Comment (ParsonsRD) [inline ctapipe/coordinates/angular_frames.py]: Hi, hmm OK that makes more sense as it is quite consistent with the documentation parameters. But looking at the astropy code they don't seen to follow this scheme, e.g.

http://docs.astropy.org/en/stable/_modules/astropy/coordinates/builtin_frames/altaz.html#AltAz

Comment (MaxNoe) [inline ctapipe/coordinates/angular_frames.py]: Mmh. Then they do not follow their own rules: http://docs.astropy.org/en/stable/development/docrules.html#documenting-classes

**Comment by kosack** [comment]
Can you fix the document warnings (see travis output)?
There are a lot of `16: ERROR: Unexpected indentation` from this commit. 

Once that's done I'll merge.

**Comment by ParsonsRD** [comment]
Hmm these errors aren't super obvious to fix. Is there a way I can run this check locally to make sure I fixed it correctly?

**Comment by kosack** [comment]
You can just run "make doc" and look for warnings or errors. I have to say Sphinx is not always great at telling you where they are.

**Comment by kosack** [comment]
also, if you use a ReStructuredText aware editor, it helps a lot. PyCharm does a pretty good job of marking errors in docstrings, for example

**Comment by jacquemier** [comment]
Since this PR request, travis is failed again (documentation part).
It would be nice to remove these warnings because any other PR is now failing.

**Comment by MaxNoe** [comment]
I don't get it. The errors are issued in the instrument module. Why did this PR break it?

**Comment by MaxNoe** [comment]
It was this PR: https://github.com/cta-observatory/ctapipe/pull/242
Not this one.

**Pull request merged**